woocommerce/woocommerce
# Add "Create a new campaign" modal

### All Submissions:


### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34909. Previous draft PR and discussions are in PR https://github.com/woocommerce/woocommerce/pull/36130.

This PR implements the "Create a new campaign" modal.

![image](https://user-images.githubusercontent.com/417342/209197516-ec6ccf0e-3829-4de1-b550-b82f3d171af6.png)

![image](https://user-images.githubusercontent.com/417342/209197676-863e91fd-e2f8-4a5e-919a-ef8b34473702.png)

The implemented modal here is different from the figma designs in https://github.com/woocommerce/woocommerce/issues/34909. Specifically, there are no borders within the modal component. This is to be more consistent with the [WordPress Modal component](https://wordpress.github.io/gutenberg/?path=/story/components-modal--default) which no longer has border under the modal title since PR https://github.com/WordPress/gutenberg/pull/40781.

The new campaign types data is loaded from `GET /marketing/campaign-types` API. The API is called when the Campaigns card is rendered, so that when users click on the "Create new campaign" button, the data is already there and no additional waiting is required.

### How to test the changes in this Pull Request:




Pre-conditions: 

- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features page: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
- Installed and activated a marketing channel, such as: 
	- Google Listings and Ads (Note: you may need to use a filter, see PR https://github.com/woocommerce/google-listings-and-ads/pull/1904).
	- Pinterest for WooCommerce (see PR https://github.com/woocommerce/pinterest-for-woocommerce/pull/705).

1. Go to WooCommerce Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
2. Click on the "Create new campaign" button in the Campaigns card.
3. You should see the modal like the screenshot above.

### Other information:


### FOR PR REVIEWER ONLY:
